client/mm/exchange_adaptor.go

```diff
@@ -2565,15 +2485,22 @@ func (u *unifiedExchangeAdaptor) updatePendingDEXOrder(o *core.Order) {
 	}
 
 	pendingOrder.txsMtx.Lock()
-	u.checkPendingDEXOrderTxs(pendingOrder, o)
+	effects := u.checkPendingDEXOrderTxs(pendingOrder, o)
+	var havePending bool
+	for _, v := range effects.pending {
+		if v > 0 {
+			havePending = true
+			break
+		}
+	}
 	pendingOrder.txsMtx.Unlock()
```

Comment:
We were marking orders as complete before the redeem tx was confirmed, so the value was still in `effects.pending` and we weren't counting it below.